api_app/analyzers_manager/observable_analyzers/spyse.py

```diff
@@ -0,0 +1,57 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+import re
+from api_app.exceptions import AnalyzerRunException, AnalyzerConfigurationException
+from api_app.analyzers_manager import classes
+
+from tests.mock_utils import if_mock_connections, patch, MockResponse
+
+
+class Spyse(classes.ObservableAnalyzer):
+    base_url: str = "https://api.spyse.com/v4/data/"
+
+    def set_params(self, params):
+        self.analysis_type = params.get("spyse_analysis", "search")
+        self.__api_key = self._secrets["api_key_name"]
+
+    def run(self):
+        if self.analysis_type == "search":
+            params = {"key": self.__api_key, "minify": True}
+
+
+        if self.observable_classification == self.ObservableTypes.DOMAIN:
+            uri = f"domain/{self.observable_name}"
+        elif self.observable_classification == self.ObservableTypes.IP:
+            uri = f"ip/{self.observable_name}"
+        elif self.observable_classification == self.ObservableTypes.GENERIC:
+            if re.match(r"^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$", self.observable_name):
+                uri = f"email/{self.observable_name}"
```

Comment:
What if the type is `generic` and the value is not an email address? You can raise an `AnalyzerRunException` for this.